packages/caliper-core/test/worker/caliper-worker.js

```diff
@@ -161,5 +167,302 @@ describe('Caliper worker', () => {
+                mockTestMessage.getNumberOfTxs.returns(numberOfTxs);
+                mockTestMessage.getRoundDuration.returns(null);
+
+                let submittedTx = 0;
+
+                mockWorkload.submitTransaction.resetHistory();
+
+                // Stub the methods
+                mockStats.getTotalSubmittedTx.callsFake(() => submittedTx);
+                mockStats.getTotalFinishedTx.callsFake(() => submittedTx)
+
+
+                mockWorkload.submitTransaction.callsFake(async () => {
+                    submittedTx += 1;
+                    if (submittedTx === 3) {
+                        return Promise.reject(new Error('Transaction submission failed'));
+                    }
+                    return Promise.resolve();
+                });
+
+                await worker.executeRound(mockTestMessage).should.be.rejectedWith('Transaction submission failed');
+
+                // Ensure transactions stop after the error and resources are cleaned up
+                sinon.assert.callCount(mockWorkload.submitTransaction, 4);
+                sinon.assert.calledOnce(deactivateMethod);
+                sinon.assert.calledOnce(mockRate.end);
+                sinon.assert.calledOnce(mockWorkload.cleanupWorkloadModule);
+                sinon.assert.called(mockConnector.releaseContext);
+
+                mockTestMessage.getRoundDuration.reset();
+                mockWorkload.submitTransaction.reset();
+                mockStats.getTotalFinishedTx.reset();
+            });
+        });
+
+        it('should execute the round for a specified duration', async function() {
+            this.timeout(5000);
+
+            const worker = new CaliperWorker(mockConnector, 1, mockMessenger, 'uuid');
+            await worker.prepareTest(mockTestMessage);
+
+            mockWorkload.submitTransaction.resolves();
+
+            mockTestMessage.getRoundDuration.returns(1);
+            mockTestMessage.getNumberOfTxs.returns(null);
+
+            const startTime = Date.now();
+            const mockStats = {
+                getRoundStartTime: () => startTime,
+                getTotalSubmittedTx: sinon.stub(),
+                getTotalFinishedTx: sinon.stub(),
+                getCumulativeTxStatistics: sinon.stub().returns({})
+            };
+            worker.internalTxObserver.getCurrentStatistics = () => mockStats;
+
+            await worker.executeRound(mockTestMessage);
+
+            const endTime = Date.now();
+            const elapsedTime = endTime - startTime;
+
+            elapsedTime.should.be.within(950, 1100);
+
+            const callCount = mockWorkload.submitTransaction.callCount;
+            callCount.should.be.greaterThan(0);
+
+            sinon.assert.calledOnce(deactivateMethod);
+            sinon.assert.calledOnce(mockWorkload.cleanupWorkloadModule);
+            sinon.assert.called(mockConnector.releaseContext);
+        });
+
+        it('should handle errors during a duration-based round', async function() {
```

Comment:
Be more explicit here about what should happen when an error occurs. These unit tests are basically the documentation of the internal behaviour of the code base.